components/script/dom/htmlinputelement.rs

```diff
@@ -860,6 +871,28 @@ impl HTMLInputElement {
             target.fire_bubbling_event(atom!("change"));
         }
     }
+
+    // https://html.spec.whatwg.org/multipage/#value-sanitization-algorithm
+    fn sanitize_value(&self) {
+        let new_value = match self.type_() {
+            atom!("text") | atom!("search") | atom!("tel") | atom!("password") => {
+                DOMString::from(self.textinput.borrow().get_content()
+                                                       .chars()
+                                                       .filter(|&c| c != '\r' && c != '\n')
+                                                       .collect::<String>())
+            }
+            atom!("url") => {
+                DOMString::from(self.textinput.borrow().get_content()
+                                                       .chars()
+                                                       .filter(|&c| c != '\r' && c != '\n')
+                                                       .collect::<String>()
+                                                       .trim_matches(char::is_whitespace))
```

Comment:
This is wrong, that should trim only ASCII whitespace.

Reply:
Your suggestion doesn't match up with the spec: https://html.spec.whatwg.org/multipage/input.html#url-state-(type=url).

Reply:
What?

> The value sanitization algorithm is as follows: Strip newlines from the value, then strip leading and trailing ASCII whitespace from the value.